### ISSUE TITLE
problem using the converter for a given pom to YAML

### ISSUE THREAD

**Issue by jmtd** [takari/polyglot-maven#99]
Attempting to convert [this pom](https://github.com/jmtd/wadc/blob/master/pom.xml):

> [ERROR] Failed to execute goal io.takari.polyglot:polyglot-translate-plugin:0.1.18:translate (default-cli) on project wadc: Execution default-cli of goal io.takari.polyglot:polyglot-translate-plugin:0.1.18:translate failed: No JavaBean properties found in java.io.File -> [Help 1]

If -e or -X output would be helpful please let me know; I have very little experience of Maven.
- Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-10T16:41:47+00:00)
- java version "1.8.0_45"

Thanks!

**Comment by InfoSec812** [comment]
Same here... 

```
[ERROR] Failed to execute goal io.takari.polyglot:polyglot-translate-plugin:0.1.19:translate (default-cli) on project logging: Execution default-cli of goal io.takari.polyglot:polyglot-translate-plugin:0.1.19:translate failed: No JavaBean properties found in java.io.File -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.takari.polyglot:polyglot-translate-plugin:0.1.19:translate (default-cli) on project logging: Execution default-cli of goal io.takari.polyglot:polyglot-translate-plugin:0.1.19:translate failed: No JavaBean properties found in java.io.File
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal io.takari.polyglot:polyglot-translate-plugin:0.1.19:translate failed: No JavaBean properties found in java.io.File
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:145)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	... 20 more
Caused by: org.yaml.snakeyaml.error.YAMLException: No JavaBean properties found in java.io.File
	at org.sonatype.maven.polyglot.yaml.ModelRepresenter.getProperties(ModelRepresenter.java:216)
	at org.sonatype.maven.polyglot.yaml.ModelRepresenter.access$300(ModelRepresenter.java:43)
	at org.sonatype.maven.polyglot.yaml.ModelRepresenter$RepresentJavaBean.representData(ModelRepresenter.java:62)
	at org.yaml.snakeyaml.representer.BaseRepresenter.representData(BaseRepresenter.java:107)
	at org.sonatype.maven.polyglot.yaml.ModelRepresenter.representJavaBean(ModelRepresenter.java:162)
	at org.sonatype.maven.polyglot.yaml.ModelRepresenter.access$400(ModelRepresenter.java:43)
	at org.sonatype.maven.polyglot.yaml.ModelRepresenter$RepresentJavaBean.representData(ModelRepresenter.java:66)
	at org.yaml.snakeyaml.representer.BaseRepresenter.representData(BaseRepresenter.java:107)
	at org.yaml.snakeyaml.representer.BaseRepresenter.represent(BaseRepresenter.java:60)
	at org.yaml.snakeyaml.Dumper.dump(Dumper.java:51)
	at org.yaml.snakeyaml.Yaml.dumpAll(Yaml.java:140)
	at org.yaml.snakeyaml.Yaml.dump(Yaml.java:128)
	at org.sonatype.maven.polyglot.yaml.YamlModelWriter.write(YamlModelWriter.java:46)
	at org.sonatype.maven.polyglot.io.ModelWriterSupport.write(ModelWriterSupport.java:47)
	at org.sonatype.maven.polyglot.io.ModelWriterSupport.write(ModelWriterSupport.java:36)
	at org.sonatype.maven.polyglot.TeslaModelTranslator.translate(TeslaModelTranslator.java:70)
	at org.sonatype.maven.polyglot.TeslaModelTranslator.translate(TeslaModelTranslator.java:56)
	at io.takari.maven.polyglot.TranslatorMojo.execute(TranslatorMojo.java:51)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	... 21 more
```

**Comment by johanfrick-mobenga** [comment]
I have the same problem :-(

**Comment by johanfrick-mobenga** [comment]
A bug seems to have been introduced in version 0.1.15. With 0.1.14 it works

**Comment by jmtd** [comment]
I've just taken a look at my extensions.xml file and I was hitting this problem with 0.1.12.

**Comment by johanfrick-mobenga** [comment]
I don't think the version in extensions.xml matter. To me it seems like it only looks at the version you type in the command. E.g.:
mvn io.takari.polyglot:polyglot-translate-plugin:**0.1.14**:translate   -Dinput=pom.xml -Doutput=pom.yml

**Comment by jmtd** [comment]
Sorry, you're right, I'm getting mixed up between translation and running mvn with a translated POM. I'm now getting a InternalErrorException in either case, so I need to figure that out before I can get back to figuring this out :)

**Comment by lefou** [comment]
Is this issue still valid with latest release (0.2.1)?

**Comment by sarxos** [comment]
Hi @lefou, I tested POM given in the first post and was able to convert it to yaml with (almost) no problems. The only issue is the fact that after pom.yaml is generated it contains line at the end of file which contains this:

```
pomFile: {}
```

It has to be removed.

```plain
$ wget https://raw.githubusercontent.com/jmtd/wadc/master/pom.xml
$ mvn io.takari.polyglot:polyglot-translate-plugin:translate -Dinput=pom.xml -Doutput=pom.yaml
```

Output:

```plain
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building WadC 2.2
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- polyglot-translate-plugin:0.2.1:translate (default-cli) @ wadc ---
[INFO] Translating /home/sarxos/workspace-neon/test-polyglot-2/pom.xml -> /home/sarxos/workspace-neon/test-polyglot-2/pom.yaml
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 0.431 s
[INFO] Finished at: 2018-01-16T19:28:32+01:00
[INFO] Final Memory: 9M/236M
[INFO] ------------------------------------------------------------------------
```

And generated yaml:

```yaml
modelVersion: 4.0.0
groupId: org.redmars.wadc
artifactId: wadc
version: 2.2
packaging: jar
name: WadC
url: https://jmtd.net/wadc/
properties: {project.build.sourceEncoding: UTF-8}
build:
  plugins:
  - groupId: org.apache.maven.plugins
    artifactId: maven-compiler-plugin
    version: 3.1
    configuration: {source: 8, target: 8}
  - groupId: org.apache.maven.plugins
    artifactId: maven-jar-plugin
    version: 2.4
    configuration:
      archive:
        manifest: {mainClass: org.redmars.wadc.WadCLaunch}
  - groupId: pl.project13.maven
    artifactId: git-commit-id-plugin
    version: 2.1.15
    configuration:
      generateGitPropertiesFile: true
      generateGitPropertiesFilename: src/main/resources/git.properties
      includeOnlyProperties: {includeOnlyProperty: ^git.commit.id.describe$}
    executions:
    - goals: [revision]
      id: default
      inherited: true
      priority: 0
  - groupId: org.apache.maven.plugins
    artifactId: maven-assembly-plugin
    version: 2.5.3
    configuration: {descriptor: src/assembly/zip.xml}
    executions:
    - goals: [single]
      id: create-archive
      inherited: true
      phase: package
      priority: 0
  - groupId: org.asciidoctor
    artifactId: asciidoctor-maven-plugin
    version: 1.5.2
    configuration: {sourceDirectory: '${basedir}/doc', outputDirectory: '${project.build.directory}/docs',
      backend: html, imagesDir: ./}
    executions:
    - goals: [process-asciidoc]
      id: output-html
      inherited: true
      phase: generate-resources
      priority: 0
pomFile: {}
```

Just make sure to **remove last line**. It breaks the pom.

**Comment by mosabua** [comment]
Fixed wit 0.3.1
